Feature/chung step2

kochungcheon opened the pull request:
[FolderMetadata]
- 폴더 속 폴더를 위해 부모 폴더 레코드 생성
- 폴더와 파일 구분을 위해 타입 레코드 생성
- 자주 사용되는 부모 폴더id는 인덱스 지정

[폴더 생성]
- 부모 폴더가 널 일 경우 루트 폴더에 폴더가 생성되도록 구현

[폴더 조회]
- 개별 유저의 폴더 및 파일 조회 가능하도록 구현, 루트 폴더에는 여러 유저의 폴더와 파일이 있을 수 있기 때문입니다
- 커서 기반 페이지 네이션을 도입했습니다. 폴더 조회 시 하위 폴더/파일이 많을 경우  데이터 로드에 많은 시간이 걸릴 수 있기 때문입니다. 오프셋 기반은 매 요청 마다 데이터 첫 행부터 검색해야 해서 성능적으로 아쉬워 커서 기반 페이지 네이션을 선택하였습니다 

[파일 로직 변경]
- 폴더 기능 도입에 따른 변경
- 부모 폴더 레코드 추가
- 파일 조회 시 부모 폴더가 널 일 경우 처리

kochungcheon committed:
refactor: FileService 의존 관계 변화

- FileReader. FileWriter 의존 변화 (기존 dataAccessService)

- 사유: DIP

kochungcheon committed:
feat: 폴더 생성

[FolderController]
- createFolder
성공 시 201 반환

[FolderService]
- 인터페이스에 의존
- createBy
부모 폴더는 null 이 될 수 있다
null 일 경우 스토리지 최상단 폴더 아래 생성

[FolderReader]
- findPathBy
- findBy
- verifyBy
- checkDuplicateBy

[FolderWriter]
- persist

[FolderDataAccessService]
- FolderReader 구현 클래스
- FolderWriter 구현 클래스

[ErrorCode]
- FOLDER_CREATE_ERROR
- UNAUTHORIZED_FOLDER_ACCESS
- DUPLICATE_FOLDER_NAME
- CANNOT_FOUND_FOLDER

[FolderRepository]
- existsByParentIdAndUserId
- existsByParentIdAndUserIdAndOriginalFolderName
인덱스인 ParentId를 활용

[FolderMetadata]
- 인덱스 설정
자주 사용되는 ParentId를 인덱스로 설정
- 빌더 패턴 사용
인자 개수에 따른 대응

[FolderUtil]
- createFolder
폴더 생성 메서드

[테스트 코드 작성]
- 기능 개발 따른 테스트 코드 작성

kochungcheon committed:
feat: 폴더 이름 변경

[FolderController]
- changeFolderNameBy
성공 시 204 반환
Patch : 데이터 일부 변환

[FolderService]
- changeFolderNameBy
폴더 이름 변경

[FolderWriter]
- changeFolderNameBy

[FolderDataAccessService]
- validateFolderOwnershipBy
폴더에 접근 권한 있는 유저인지 확인

- changeFolderNameBy(String folderName, Long folderId, Long userId)
validateFolderOwnershipBy, changeFolderNameBy 로 폴더 이름 변경 로직 수행

- changeFolderNameBy(String folderName, Long folderId)
폴더 이름 변경 후 저장

[ErrorCode]
- FOLDER_CREATE_ERROR
- UNAUTHORIZED_FOLDER_ACCESS
- DUPLICATE_FOLDER_NAME
- CANNOT_FOUND_FOLDER

[FolderRepository]
- existsByIdAndUserId

[요청 DTO 생성]
- FolderNameChangeRequest

[테스트 코드 작성]
- 기능 개발 따른 테스트 코드 작성

kochungcheon committed:
feat: MetadataType

- file, folder 구분 목적

kochungcheon committed:
feat: step2

[FileDataAccessService]
- 클래스 내부에서만 쓰이는 메서드 접근 제어자 private으로 변경

[FileMetadata]
- @Builder 적용 인자 개수 증가로 변경

[FileRepository]
- checkDuplicateFileName 부모 Id 널일때 고려

[FileService]
- findChildBy 파일 조회 메서드

[FolderDataAccessService]
- 클래스 내부에서만 쓰이는 메서드 접근 제어자 private으로 변경

[FolderMetadata]
- parentId에 인덱스 적용

[FolderReader]
- FolderReader 생성

[FolderRepository]
- existsByParentIdAndUploaderIdAndOriginalFolderName 생성
- existsByIdAndUploaderId 생성
- findByParentIdAndUploaderId 생성
- 전부 pk나 세컨더리 인덱스를 타게 유도

[FolderService]
- findPathBy MetaService 에서도 활용하기 때문에 접근 제어자 public
- findChildBy 생성

[MetadataController]
- MetadataService의 컨트롤러

[MetadataService]
- 파사드 패턴 적용 FolderService, FileService 의존성 관리
- uploadFile 생성
- uploadFile 생성

[기타]
- 전송, 수신 Dto 생성
- 변경 사항에 따른 테스트 코드 추가

VSFe commented on gradlew.bat:
```diff
@@ -2,7 +2,7 @@
 @rem
 @rem Licensed under the Apache License, Version 2.0 (the "License");
-@rem you may not use this metadata except in compliance with the License.
+@rem you may not use this fileMetadata except in compliance with the License.
```
?

kochungcheon replied:
아 기존 metadata를 전체 바꿈 단축기로 fileMetadata 로 바꿨었는 데 그것때문에 발생만 문제 같습니다
다시 push 하겠습니다

kochungcheon committed:
chore: gradlew.bat 수정
- fileMetadata -> metadata

VSFe commented on src/main/java/com/c4cometrue/mystorage/folder/FolderDataAccessService.java:
```diff
@@ -0,0 +1,75 @@
+		}
+	}
+
+	// 적절한 예외 던져라
```
네

kochungcheon committed:
refactor: 트랜잭션 범위 및 코드 위치 변경

- 트랜잭션 범위를 저장 로직/ 삭제 로직에 한정했습니다.

  기존의 방식은 IO 작업(실제 파일을 업로드, 삭제)으로 인해 트랜잭션이 오래 열려있게 되기 때문입니다.

- I0 작업과 DB 저장 작업 순서를 바꿨습니다.

  IO 은 취소할 수 없는 작업이기 때문입니다. 기존 방식에 따르면 데이터 불일치 문제가 일어날 수 있어 DB 작업 이후 IO 작업을 하게 구성하였습니다

pear96 commented on src/main/java/com/c4cometrue/mystorage/dto/FileUploadRequest.java:
```diff
@@ -2,12 +2,12 @@
-								@NotNull(message = "사용자 id는 null 될 수 없습니다") long userId) {
-	public static FileUploadRequest of(MultipartFile multipartFile, long userId) {
-		return new FileUploadRequest(multipartFile, userId);
+public record FileUploadRequest(@NotNull(message = "파일을 전달해주세요") MultipartFile multipartFile,
```
`@NotNull`로 바꾸셨군요! 그런데 저는 테스트해보면 다른 입력값은 `MethodArgumentNotValidException`가 발생하는데 Multipartfile은 파일을 담지 않고 보내면 `ConstraintViolationException`가 발생하더라구요. 예외 핸들러에서 한번에 처리해줄 수 없는데 청천님은 어떻게 하려고 하셨나요..

kochungcheon replied:
> `@NotNull`로 바꾸셨군요! 그런데 저는 테스트해보면 다른 입력값은 `MethodArgumentNotValidException`가 발생하는데 Multipartfile은 파일을 담지 않고 보내면 `ConstraintViolationException`가 발생하더라구요. 예외 핸들러에서 한번에 처리해줄 수 없는데 청천님은 어떻게 하려고 하셨나요..



> `@NotNull`로 바꾸셨군요! 그런데 저는 테스트해보면 다른 입력값은 `MethodArgumentNotValidException`가 발생하는데 Multipartfile은 파일을 담지 않고 보내면 `ConstraintViolationException`가 발생하더라구요. 예외 핸들러에서 한번에 처리해줄 수 없는데 청천님은 어떻게 하려고 하셨나요..

예외에 많이 미진해서 하은님 코드 보며 많이 배우고 있습니다!

pear96 commented on src/main/java/com/c4cometrue/mystorage/dto/FolderContentsRequest.java:
```diff
@@ -0,0 +1,12 @@
+
+public record FolderContentsRequest(
+	Long folderId,
+	@NotNull(message = "사용자 id는 널이 될 수 없습니다") long userId
```
왜 `Long`과 `long` 으로 구분지으셨는지 궁금합니다.. 조회용 DTO같은데 folderId는 null이어도 되나요?

kochungcheon replied:
폴더 id 가 null 일 시 최상단 부모를 의미하게 했습니다! 그리고 userId 가 null 이 될 수 없게 했기 때문에 @NotNull 과 long 을 썼습니다

pear96 commented on src/main/java/com/c4cometrue/mystorage/dto/FolderContentsRequest.java:
```diff
@@ -0,0 +1,12 @@
+	Long folderId,
+	@NotNull(message = "사용자 id는 널이 될 수 없습니다") long userId
+) {
+	public static FolderContentsRequest of(Long folderId, long userId) {
```
이 DTO를 생성하는 곳에서 `new FolderRequest()` 를 하지 않으려고 `of`로 만드신걸까요? 만약 그렇다면 더 좋은 점이 뭐죠?

kochungcheon replied:
https://tecoble.techcourse.co.kr/post/2020-05-26-static-factory-method/
메서드 명 가지는 장점이 있습니다. 메서드 내에서 제가 자유롭게 에러 코드를 던진다던가 다양하게 생성을 할 수 있다는 행위가 가능해서 new 보다는 of를 선호합니다

pear96 commented on src/main/java/com/c4cometrue/mystorage/dto/FolderNameChangeRequest.java:
```diff
@@ -0,0 +1,16 @@
+
+public record FolderNameChangeRequest(
+	@NotNull(message = "유저id는 널이 될 수 없습니다") long userId,
+	Long folderId,
```
폴더의 이름을 바꾸는 경우에도 Null일 수 있나요?

kochungcheon replied:
오 아닙니다! 수정하겠습니다!

pear96 commented on src/main/java/com/c4cometrue/mystorage/folder/FolderDataAccessService.java:
```diff
@@ -0,0 +1,75 @@
+
+@Service
+@RequiredArgsConstructor
+public class FolderDataAccessService implements FolderReader, FolderWriter {
```
Repository를 FolderService가 아닌 DataAccessService에서 의존하며 분리하신 이유가 있을까요?

kochungcheon replied:
비지니스로직과 데이터를 처리하는 로직을 분리하고 싶었습니다!

pear96 commented on src/main/java/com/c4cometrue/mystorage/file/FileController.java:
```diff
@@ -20,12 +18,6 @@
 public class FileController {
 	private final FileService fileService;
 
-	@PostMapping
```
Metadata 컨트롤러로 인해 파일 업로드를 이동하셨군요..
그리고 메타데이터 서비스가 파일과 폴더 서비스를 다시 의존하고 있군요..

kochungcheon replied:
파일과 폴더 서비스에 의존성이 생기는 게 자연스러운가 생각해서 
Metadata 서비스를 만들어서 의존을 관리하고자 했습니다
지금 생각해보니 
네이밍이 MetadataFasade 쪽으로 됐어야 하고 
파일과 폴더 사이 의존에 대해 좀 더 생각해봐야겠네요

pear96 commented on src/main/java/com/c4cometrue/mystorage/folder/FolderController.java:
```diff
@@ -0,0 +1,32 @@
+@RestController
+@RequestMapping("/folders")
+@RequiredArgsConstructor
+public class FolderController {
```
생각해보니 유효성 검사 어노테이션이 없군요.. 유효성 검사에 문제가 없었나요?

kochungcheon replied:
> 생각해보니 유효성 검사 어노테이션이 없군요.. 유효성 검사에 문제가 없었나요?

이 부분도 위에서 언급했 듯 많이 배우고 있습니다 감사합니다!

kochungcheon replied:
> 생각해보니 유효성 검사 어노테이션이 없군요.. 유효성 검사에 문제가 없었나요?

@Vaild 생각보다 재밌는 내용이 많군요 감사합니다!

pear96 commented on src/main/java/com/c4cometrue/mystorage/file/FileService.java:
```diff
@@ -2,52 +2,59 @@
-		Path path = Paths.get(storagePath, storedFileName);
-		Metadata metadata = Metadata.of(originalFileName, storedFileName, path.toString(), userId);
+		String storedFileName = FileMetadata.storedName();
+		Path path = Paths.get(basePath, storedFileName);
```
헐 저도 `get`으로 바꿔야겠다.... 계속 `resolve()` 해주고 있었는데...

pear96 commented on src/main/java/com/c4cometrue/mystorage/folder/FolderController.java:
```diff
@@ -0,0 +1,32 @@
+	@PostMapping
+	@ResponseStatus(HttpStatus.CREATED)
+	public void createFolder(FolderCreateRequest req) {
+		folderService.createBy(req.userId(), req.userFolderName(), req.parentId());
```
(정답은 모름 그냥 같이 고민하자는 뜻)
Service에 DTO를 바로 넘기는 것과 parsing해서 넘기는 것 중 어느게 더 좋은거란게 있을까요?

kochungcheon replied:
> (정답은 모름 그냥 같이 고민하자는 뜻) Service에 DTO를 바로 넘기는 것과 parsing해서 넘기는 것 중 어느게 더 좋은거란게 있을까요?

그 부분에 대해 많은 의견이 갈리는 거 같더라고요
현재는 컨트롤러 까지 받아오는 걸 DTO 로 하고 있습니다. parsing 했을 때 장점은 명확하게 인자가 보이니 (ex. add(int a, int b) )Service 레이어에서 인자들을 보면서 작업하기 용이 하다는 것을 들 수 있을 거 같아요 DTO 를 통째로 넘기면 어떤 장점이 있을 까요

VSFe commented on src/main/java/com/c4cometrue/mystorage/meta/MetadataService.java:
```diff
@@ -0,0 +1,45 @@
+
+@Service
+@RequiredArgsConstructor
+public class MetadataService {
```
굳이 분리할 필요가 있나하는 생각이 드네요.
MetadataService가 아닌 File 및 FolderService에도 메타데이터를 다루고 있지 않나요?

kochungcheon replied:
폴더 서비스와 파일 서비스에서 발생한 의존성을 별도로 관리하고 싶어 만들었습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/meta/MetadataController.java:
```diff
@@ -0,0 +1,33 @@
+public class MetadataController {
+	private final MetadataService metadataService;
+	@GetMapping("/metadata")
+	public ResponseEntity<List<MetaResponse>> getFolderContents(FolderContentsRequest req) {
```
역할이 정확히 이해가 안 가네요.
폴더 목록 조회는 FolderController로 가는게 더 자연스럽지 않을까요?

kochungcheon replied:
이 부분에서 고민을 좀 많이 해봤는 데 
파일과 폴더는 의존성이 생길 수 밖에 없는 관계네요 의존성 형성하는 방향으로 가고 
컨트롤러 위치도 그에 맞게 수정하겠습니다

VSFe commented on src/main/java/com/c4cometrue/mystorage/folder/FolderWriter.java:
```diff
@@ -0,0 +1,7 @@
+package com.c4cometrue.mystorage.folder;
+
+public interface FolderWriter {
```
2개 이상의 구현체가 있지 않은 상황에서, 굳이 Interface를 만들어야 할지에 대한 의문이 들어요.
특히나 Reader, Writer 모두 하나의 Service로 때려박은 상황에선 더더욱요.

kochungcheon replied:
이 부분은 많이 고민 됩니다 
결합도를 낮출수있어 가져오는 장점과 Reader, Writer 를 통해 해당 메서드가 어떤 역할을 하는지  알려줄수있다는 장점에서요 
흡 좀 더 고민 해보겠습니다!

kochungcheon replied:
> 2개 이상의 구현체가 있지 않은 상황에서, 굳이 Interface를 만들어야 할지에 대한 의문이 들어요. 특히나 Reader, Writer 모두 하나의 Service로 때려박은 상황에선 더더욱요.

인터페이스 제거하였습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/folder/FolderController.java:
```diff
@@ -0,0 +1,32 @@
+	@PostMapping
+	@ResponseStatus(HttpStatus.CREATED)
+	public void createFolder(FolderCreateRequest req) {
+		folderService.createBy(req.userId(), req.userFolderName(), req.parentId());
```
~By 라는 네이밍을 사용한 이유가 있을까요?

kochungcheon replied:
뒤의 인자들로 만들어진다 라는 의미였는 데 
이곳에서는 createFolder 라고 표현하는 게 더 좋을 거 같습니다

VSFe commented on src/main/java/com/c4cometrue/mystorage/file/FileRepository.java:
```diff
@@ -1,19 +1,26 @@
+public interface FileRepository extends JpaRepository<FileMetadata, Long> {
+	Optional<FileMetadata> findByIdAndUploaderId(Long id, Long uploaderId);
 
 	@Query("SELECT CASE WHEN COUNT(m) > 0 "
```
JDK 17에서는 """ 으로 Text Block 생성이 가능합니다.

kochungcheon replied:
오 감사합니다

VSFe commented on src/main/java/com/c4cometrue/mystorage/file/FileRepository.java:
```diff
@@ -1,19 +1,26 @@
-		+ "WHERE m.uploaderId = :uploaderId "
+		+ "FROM FileMetadata m "
+		+ "WHERE "
+		+ "(m.parentId = :parentId OR (m.parentId IS NULL AND :parentId IS NULL)) "
```
(parent_id, uploader_id, original_file_name) 의 인덱스가 필요하겠네요.

kochungcheon replied:
넵 추가하겠습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/file/FileDataAccessService.java:
```diff
@@ -1,41 +1,57 @@
+		}
+	}
+
+	public FileMetadata findBy(Long fileId, Long userId) {
```
existBy랑 매칭이 좀 어색해보이네요.
- existBy는 fileId로 Unique 하게 찾을 수 있어보이는데, findBy는 userId를 받는 이유가 있을까요?

kochungcheon replied:
악성 유저가 다른 사람의 파일에 접근하는 것을 막고자 의도 했습니다!

kochungcheon replied:
delete의 경우 FileService 에서 유저가 해당 파일 소유주인지 확인을 한 다음 오기 때문에 별도로 userId 는 추가 하지 않았습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/file/FileDataAccessService.java:
```diff
@@ -1,41 +1,57 @@
-	public void duplicateBy(String fileName, Long userId) {
-		if (fileRepository.checkDuplicateFileName(fileName, userId)) {
-			throw ErrorCode.DUPLICATE_FILE_NAME.serviceException("fileName : {}", fileName);
+	public List<FileMetadata> findChildBy(Long parentId, Long userId) {
```
fileId는 Unique 하게 만들었으면서, folderId는 Unique하게 만들지 않은 이유가 있나요?

kochungcheon replied:
좀더 설명 부탁드려도 되나요!? 
fileId는 Unique 라는 부분을 이해못했습니다..

VSFe commented on src/main/java/com/c4cometrue/mystorage/folder/FolderMetadata.java:
```diff
@@ -0,0 +1,62 @@
+
+	@Column(nullable = false)
+	@Enumerated(EnumType.STRING)
+	private MetadataType metadataType;
```
file과 folder의 테이블을 분리했으니, MetadataType 칼럼이 존재할 필요는 없어보이네요.

kochungcheon replied:
앗..!! 그렇습니다! 수정하겠습니다

VSFe commented on src/main/java/com/c4cometrue/mystorage/folder/FolderRepository.java:
```diff
@@ -0,0 +1,14 @@
+
+	boolean existsByIdAndUploaderId(Long folderId, Long userId);
+
+	List<FolderMetadata> findByParentIdAndUploaderId(Long parentId, Long userId);
```
(parent_id, user_id) 에 대한 인덱스는 필요하겠네요.

kochungcheon replied:
오 복합키.! 도입하겠습니다 그리고 이걸로 바꾸었을때 어떤 장점있을 지도 한번 파악해보겠습니다

VSFe commented on src/main/java/com/c4cometrue/mystorage/meta/MetadataService.java:
```diff
@@ -0,0 +1,45 @@
+		List<FolderMetadata> folderResponses = folderService.findChildBy(folderId, userId);
+		List<FileMetadata> fileResponses = fileService.findChildBy(folderId, userId);
+
+		return Stream.concat(
```
파일이 4700개이고, 폴더가 5300개라면 어떻게 될까요?

kochungcheon replied:
아아..!!! 다 보여 줄 필요가 없군요 고민해보겠습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/util/FolderUtil.java:
```diff
@@ -0,0 +1,21 @@
+
+	public static void createFolder(Path path) {
+		try {
+			Files.createDirectories(path);
```
폴더 생성이 실패하는 케이스가 무엇이 있을까요?
굳이 물리적인 파일시스템 장애가 아니더라도, 지금 로직으로는 충분히 폴더 생성이 실패할 수 있어요.

kochungcheon replied:
아아 폴더 경로가 중복될 수 있군요 UUID 로 대비된다 생각했는 데 
UUID 로 생성된 폴더 경로가 중복일 경우 다시 생성 하는 로직이 필요하겠네요

VSFe commented on src/main/resources/application.yml:
```diff
@@ -10,6 +10,7 @@ spring:
       hibernate:
         dialect: org.hibernate.dialect.MySQL8Dialect
+        show_sql: true
 
```
위쪽에 적힌 `ddl-auto: create`는 제거하세요.
테스트할 데이터가 쌓일 수가 없는 구조네요...

통합 테스트를 구축하기 위한 property가 아닌 이상, 저걸 사용할 이유는 없습니다.

kochungcheon replied:
넵 알겠습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/meta/MetadataController.java:
```diff
@@ -0,0 +1,33 @@
+	}
+
+	@PostMapping("/files")
+	@ResponseStatus(HttpStatus.CREATED)
```
물리적인 파일 업로드랑, 메타데이터 업로드를 별도로 분리한건가요?
그렇다면 왜 폴더에 대한 메타데이터 업로드 메서드는 존재하지 않는걸까요?

kochungcheon replied:
위의 코멘트 처럼 수정하겠습니다!

VSFe commented on src/main/java/com/c4cometrue/mystorage/folder/FolderDataAccessService.java:
```diff
@@ -0,0 +1,75 @@
+
+	private void changeFolderNameBy(String folderName, Long folderId) {
+		FolderMetadata folderMetadata = findBy(folderId);
+		folderMetadata.changeFolderName(folderName);
```
변경한 이름이 충돌되지 않는지는 확인하지 않네요.

kochungcheon replied:
아 UUID 로 생성 해도 충돌 될 경우가 존재하군요 ..! 수정하겠습니다!

kochungcheon replied:
> 변경한 이름이 충돌되지 않는지는 확인하지 않네요.

현재 폴더는 사용자가 저장한 이름과 서버에 저장용 이름 이렇게 두개로 저장됩니다 
 changeFolderNameBy 라는 메소드는 사용자가 저장한 이름을 바꾸는 메소드 입니다
이 경우에도 변경 이름이 충돌되는 지 검사 하는 게 좋을 까요!?

VSFe reviewed (changes requested):
고생하라는 의미에서 (?) 리뷰 난이도를 올려봤습니다.

pear96 commented on src/test/java/com/c4cometrue/mystorage/file/FileDataAccessServiceTest.java:
```diff
@@ -31,10 +34,17 @@ void shouldDeleteByFileId() {
+	@DisplayName("파일 삭제 테스트 : 실패")
+	void shouldDeleteByFileIdFail() {
+		when(fileRepository.existsById(FILE_ID)).thenReturn(false);
+		assertThrows(ServiceException.class, () -> fileDataAccessService.deleteBy(FILE_ID));
```
원하는 exception의 이름이 맞는지 확인하는 검증 단계가 있으면 좋을 것 같아요

kochungcheon replied:
오 그러면 확실히 의도한 에러가 나온 지 알 수 있는 장점이 있네요

pear96 commented on src/test/java/com/c4cometrue/mystorage/folder/FolderDataAccessServiceTest.java:
```diff
@@ -0,0 +1,117 @@
+	void getChildFolder() {
+		folderDataAccessService.findChildBy(null, USER_ID);
+
+		then(folderRepository).should(times(1)).findByParentIdAndUploaderId(null, USER_ID);
```
`verify` 대신 `then`을 쓰신 이유가 궁금합니다

kochungcheon replied:
https://brunch.co.kr/@springboot/292 부분 때문에요 사실 어떤 게 좋은 지 모르겠습니다

kochungcheon committed:
refactor: 파일 조회 컨트롤러 위치 변경
- MetadataController -> FileController
- base위치 파악 위해 FileService 에 FolderService 의존 생성
- 변경 지점에 따른 테스트 코드 수정

kochungcheon committed:
refactor: Reader, Writer 제거

- 인터페이스 제거에 따른 코드 수정 작업

- XXXDataAcessService -> XXXHandlerService 로 수정 작업

- 변화에 따른 테스트 코드 수정

kochungcheon committed:
refactor: FolderMetadata index 추가 및 제거 작업
- parentId, uploaderId, originalFolderName 을 인덱스로 생성
- parentId 에만 잡힌 인덱스 제거

kochungcheon committed:
refactor: MetadataService -> StoregeFasadeService

- 명칭 변화에 따른 코드 수정

kochungcheon committed:
chore: 오탈자 수정

kochungcheon committed:
refactor: 유효성 검사 어노테이션 추가

kochungcheon committed:
feat: 커서 기반 페이징 도입-

kochungcheon committed:
chore : 사용하지 않는 파일 정리

kochungcheon committed:
test : 변경 사항에 따른 테스트 코드 추가

- FileDataHandlerService getFileList, hashNext

- FileService getFiles

- FolderDataHandlerService getFolderList hasNext

- FolderService getFolders

- StorageFacadeService getFolderContents

- PagingUtil calculateSize createPageable

kochungcheon committed:
chore : 코드 스멜 수정
